### DISCUSSION TITLE
How to debug page fault in enclave

### DISCUSSION THREAD

**Discussion by mdrssv** [keystone-enclave/keystone#461] [General]
Hi,

I'm currently trying to write my eapp Rust which has been working fine to a degree. However when I start using crypto libraries such as [RustCrypto/sha2](https://docs.rs/sha2/latest/sha2/) or [RustCrypto/chacha20](https://docs.rs/chacha20/latest/chacha20/). I'll encounter page faults every time I try to encrypt or hash data. I've the tested my code on x86 as well in the Riscv QEMU VM without any problems however when switching to keystone I get page faults. I've managed to track these page faults down to memset which isn't very insightful imo.

Does anybody else have an idea how debug this issue or what else might be causing an page fault?

**Comment by tayibs1** [comment]
Hi, this isnt a solution but I am also working on a project that involves creating Eapps. Is there any other documentation that you used apart from the keystone enclave docs; specifically for the runtime when using Eyrie or sel4.

Thank you in advance
